docs/sphinx/applications/python/deutschs_algorithm.ipynb

```diff
@@ -13,17 +13,14 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We have a function which takes in a bit and outputs a bit. This can be represented as $f: \\{0,1\\} \\longrightarrow \\{0,1\\}$. \n",
+    "Deutsch's Algorithm is a consise demonstration of the differences in computational complexity between classical and quantum algorithms for certain problems.  For Desutch's algorithm, we begin with a function which takes in a bit and outputs a bit. This can be represented as $f: \\{0,1\\} \\longrightarrow \\{0,1\\}$. \n",
+    "The function $f$ has the property that it either constant or balanced. The goal of Deutsch's Algorithm is to determine whether our given function is constant or whether it is balanced. \n",
     "\n",
-    "The function $f$ has a property; either it is constant or balanced. \n",
+    "A constant function is one in which the outputs are the same regardless of the inputs, i.e., in the case of $f: \\{0,1\\} \\longrightarrow \\{0,1\\}$, there are are two ways in which this can occur: $f(0) = f(1) = 0$ or $f(0) = f(1) = 1$.\n",
     "\n",
-    "If constant, the outputs are the same regardless of the inputs, i.e., $f(0) = f(1) = 0$ or $f(0) = f(1) = 1$.\n",
-    "\n",
-    "If balanced, the ouputs are balanced across their possibilities, i.e,  if $f(0) = 0$ then $f(1) = 1$ or if $f(0) = 1$ then $f(1) = 0$.\n",
-    "\n",
-    "The question we would like to answer is if the function is constant or balanced. \n",
+    "A balanced function is one in which the ouputs are balanced across their possibilities, i.e,  if $f(0) = 0$ then $f(1) = 1$ or if $f(0) = 1$ then $f(1) = 0$.\n",
```

Comment:
```suggestion
    "A balanced function is defined such that the ouputs are balanced across their possibilities, i.e,  if $f(0) = 0$ then $f(1) = 1$ or if $f(0) = 1$ then $f(1) = 0$.\n",
```
"that" seems better here since the clause is not optional.